libraries/aws/jobs/src/aws_iot_jobs_serialize.c

```diff
@@ -1010,104 +1015,107 @@ static AwsIotJobsError_t _parseErrorDocument( const char * pErrorDocument,
 
-        /* InvalidRequest */
-        case 16:
+                break;
 
-            if( strncmp( "\"InvalidRequest\"", pCode, codeLength ) == 0 )
-            {
-                status = AWS_IOT_JOBS_INVALID_REQUEST;
-            }
+            /* InvalidRequest */
+            case 16:
 
-            break;
+                if( strncmp( "\"InvalidRequest\"", pCode, codeLength ) == 0 )
+                {
+                    status = AWS_IOT_JOBS_INVALID_REQUEST;
+                }
 
-        /* VersionMismatch */
-        case 17:
+                break;
 
-            if( strncmp( "\"VersionMismatch\"", pCode, codeLength ) == 0 )
-            {
-                status = AWS_IOT_JOBS_VERSION_MISMATCH;
-            }
+            /* VersionMismatch */
+            case 17:
 
-            break;
+                if( strncmp( "\"VersionMismatch\"", pCode, codeLength ) == 0 )
+                {
+                    status = AWS_IOT_JOBS_VERSION_MISMATCH;
+                }
 
-        /* ResourceNotFound, RequestThrottled */
-        case 18:
+                break;
 
-            if( strncmp( "\"ResourceNotFound\"", pCode, codeLength ) == 0 )
-            {
-                status = AWS_IOT_JOBS_NOT_FOUND;
-            }
-            else if( strncmp( "\"RequestThrottled\"", pCode, codeLength ) == 0 )
-            {
-                status = AWS_IOT_JOBS_THROTTLED;
-            }
+            /* ResourceNotFound, RequestThrottled */
+            case 18:
 
-            break;
+                if( strncmp( "\"ResourceNotFound\"", pCode, codeLength ) == 0 )
+                {
+                    status = AWS_IOT_JOBS_NOT_FOUND;
+                }
+                else if( strncmp( "\"RequestThrottled\"", pCode, codeLength ) == 0 )
+                {
+                    status = AWS_IOT_JOBS_THROTTLED;
+                }
 
-        /* TerminalStateReached */
-        case 22:
+                break;
 
-            if( strncmp( "\"TerminalStateReached\"", pCode, codeLength ) == 0 )
-            {
-                status = AWS_IOT_JOBS_TERMINAL_STATE;
-            }
+            /* TerminalStateReached */
+            case 22:
 
-            break;
+                if( strncmp( "\"TerminalStateReached\"", pCode, codeLength ) == 0 )
+                {
+                    status = AWS_IOT_JOBS_TERMINAL_STATE;
+                }
 
-        /* InvalidStateTransition */
-        case 24:
+                break;
 
-            if( strncmp( "\"InvalidStateTransition\"", pCode, codeLength ) == 0 )
-            {
-                status = AWS_IOT_JOBS_INVALID_STATE;
-            }
+            /* InvalidStateTransition */
+            case 24:
 
-            break;
+                if( strncmp( "\"InvalidStateTransition\"", pCode, codeLength ) == 0 )
+                {
+                    status = AWS_IOT_JOBS_INVALID_STATE;
+                }
 
-        default:
-            break;
+                break;
+
+            default:
+                break;
```

Comment:
Unrelated to this PR as it was already there, but having an empty `default` without a comment violates MISRA 16.4. It doesn't need to be fixed for this PR but it's only a one line change to add a comment like `/* Empty default MISRA 16.4 */`.

Reply:
No problem. I can fix it in this PR. Does that comment tell user that the empty default is intended? 
I got a better idea. Put following line to defaut:

```
status = AWS_IOT_JOBS_BAD_RESPONSE;
```